tests/queries/0_stateless/01710_projection_optimize_materialize.sql

```diff
@@ -11,4 +11,7 @@ alter table z materialize projection pp settings mutations_sync=1;
 
 SELECT name, partition, formatReadableSize(sum(data_compressed_bytes) AS size) AS compressed, formatReadableSize(sum(data_uncompressed_bytes) AS usize) AS uncompressed, round(usize / size, 2) AS compr_rate, sum(rows) AS rows, count() AS part_count FROM system.projection_parts WHERE database = currentDatabase() and table = 'z' AND active GROUP BY name, partition ORDER BY size DESC;
 
+alter table z add projection pp1 (select id, sum(c) group by id);
+alter table z materialize projections settings mutations_sync=1;
```

Comment:
Maybe we should test that indices are actually materialized? For example we can add SELECT query which will read from table with WHERE expression and can get benefits from projection and we will test that we read less rows than full scan. Here is an example where we check how many rows were read https://github.com/ClickHouse/ClickHouse/blob/f2ffd727f002702ab29dff7c2d18ceaaf8e09e6e/tests/queries/0_stateless/02906_force_optimize_projection_name.sql#L27-L33

Reply:
thanks @alesapin . done.